runner.js

```diff
@@ -2,41 +2,34 @@ const fs = require('fs').promises
 const cp = require('child_process')
 const ulid = require('ulid')
 const languages = require('./languages')
-
-const buildArgs = (config, name, file) => {
-  const { image, cmd } = config
-  return [
-    'run',
-    '--rm',
-    '--name',
-    name,
-    '-v',
-    `${process.cwd()}:/usr/src/app`,
-    '-w',
-    '/usr/src/app',
-    image,
-    cmd,
-    file
-  ]
-}
+const utils = require('./utils')
 
 module.exports = function run (language, code) {
-  const config = languages[language];
+  const config = languages[language]
   if (typeof code !== 'string' || !config) return ''
+
+  const { executeExtension, compileExtension } = config
+
   return new Promise((resolve, reject) => {
     const name = ulid.ulid()
-    const fileName = `${name}.${config.extension}`
+    const fileExtension = compileExtension || executeExtension
+    const fileName = utils.buildFileName(name, fileExtension)
+
     fs.writeFile(fileName, code).then(() => {
       let output = Buffer.alloc(0)
-      const docker = cp.spawn('docker', buildArgs(config, name, fileName))
+      const docker = cp.spawn('docker', utils.buildArgs(config, name, fileName))
```

Comment:
approved but also I think you're passing an extra arg here

Reply:
forgot to update, thanks